eks/base/eks_managed_node_group/variables.tf

```diff
@@ -0,0 +1,84 @@
+variable "cluster_auth_base64" {
+  type        = string
+  description = "value of the cluster_auth_base64"
+  default     = ""
+}
+
+variable "aws_auth_roles" {
+  type        = list(any)
+  description = "List of AWS IAM users to add to aws-auth configmap"
+  default     = []
+  # i.e
+  # default = [
+  #   {
+  #     rolearn  = "arn:aws:iam::66666666666:role/role1"
+  #     username = "role1"
+  #     groups   = ["system:masters"]
+  #   },
+  # ]
+}
+
+variable "aws_auth_users" {
+  type        = list(any)
+  description = "List of AWS IAM accounts to add to aws-auth configmap"
+  default     = []
+
+  # i.e
+  #   default = [
+  #   {
+  #     userarn  = "arn:aws:iam::66666666666:user/user1"
+  #     username = "user1"
+  #     groups   = ["system:masters"]
+  #   },
+  #   {
+  #     userarn  = "arn:aws:iam::66666666666:user/user2"
+  #     username = "user2"
+  #     groups   = ["system:masters"]
+  #   },
+  # ]
+}
+
+variable "aws_auth_accounts_ids" {
+  type        = list(string)
+  description = "List of AWS IAM accounts IDs to add to aws-auth configmap"
+  default     = []
+  # i.e
+  # default = [
+  #   "777777777777",
+  #   "888888888888",
+  # ]
+
+}
+
+variable "bootstrap_extra_args" {
+  type        = string
+  description = "Extra arguments to pass to the bootstrap script"
+  default     = <<-EOT
+        # The admin host container provides SSH access and runs with "superpowers".
+        # It is disabled by default, but can be disabled explicitly.
+        [settings.host-containers.admin]
+        enabled = false
+
+        # The control host container provides out-of-band access via SSM.
+        # It is enabled by default, and can be disabled if you do not expect to use SSM.
+        # This could leave you with no way to access the API and change settings on an existing node!
+        [settings.host-containers.control]
+        enabled = true
+
+        # extra args added
+        [settings.kernel]
+        lockdown = "integrity"
+
+        [settings.kubernetes.node-labels]
+        # todo: add label to the nodes
+        app1 = "explorer-squid"
```

Comment:
I still see these applications hard coded. Can we get these from config instead?

Reply:
sure, I have refactored it now, created blue/green environments and added argocd too. These hardcoded values are passed as variable now.